### PR TITLE
Make NCELogits easier to use, and rename XXLogits to XXDense

### PR DESCRIPTION
**Pull request by eric-haibin-lin**
## Description ##
- NCELogits is usually used together with SigmoidBCELoss. This PR update the output of NCELogits to be probably distribution ([1, 0, 0, .... 0]) instead of target class. This way the output can be fed into the loss directly. 
- Rename NCELogits/ISLogits to NCEDense/ISDense, since the output of these blocks are unnormalized predictions rather than log probabilities. Otherwise it's misleading to users

### Essentials ###
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage
- [x] Code is well-documented

### Changes ###
- [ ] Feature1, tests, (and when applicable, API doc)
- [ ] Feature2, tests, (and when applicable, API doc)

## Comments ##
- If this change is a backward incompatible change, why must this change be made.
- Interesting edge cases to note here

### CODE REVIEW

File: tests/unittest/test_sampled_logits.py [hunk 1]
```diff
@@ -26,51 +26,57 @@
...
+        true_cls_cnt = mx.nd.ones((batch_size,))
+        samples = (sampled_cls, sampled_cls_cnt, true_cls_cnt)
+        with mx.autograd.record():
+            pred, new_y = model(x, samples, y)
+            assert pred.shape == (batch_size, 1+num_sampled)
+            assert new_y.shape == (batch_size,)
+            l = loss(pred, new_y)
+        l.backward()
+        mx.nd.waitall()
+    check_is_softmax_loss(nlp.model.ISDense)
```
Comment (szha) [inline tests/unittest/test_sampled_logits.py]: while this is perfectly reasonable, the same can be done with pytest.mark.parameterize, FYI. see https://github.com/dmlc/gluon-nlp/blob/master/tests/unittest/test_sequence_sampler.py#L45-L48

Comment (eric-haibin-lin) [reply tests/unittest/test_sampled_logits.py]: I'll do it that way next time

**Comment by mli** [comment]
Job PR-286/2 is complete. 
Docs are uploaded to http://gluon-nlp-staging.s3-accelerate.dualstack.amazonaws.com/PR-286/2/index.html
